contracts/AtlasMineStakerUpgradeable.sol

```diff
@@ -675,22 +676,31 @@ contract AtlasMineStakerUpgradeable is
     }
 
     /**
-     * @notice Must be used when migrating to a new contract that changes the accounting
-     *         logic of unstaked deposits. Can be used to reset value to one that would
-     *         be in place in the cast that newly-introduced logic was always in place.
+     * @notice Used to set windows during which accrual happens, and new deposits/withdrawls
+     *         are paused.
      *
-     * @dev    Cannot be used in normal operation, will only be called once as part of
-     *         an "upgradeAndCall" contract upgrade.
-     *
-     * @param _unstakedDeposits    The new value of unstakedDeposits to set.
+     * @param windows               The list of hourly windows in pairs of tuples, e.g. [0, 1, 12, 13]
      */
-    function resetUnstakedAndStake(uint256 _unstakedDeposits) external {
-        require(!_resetCalled, "reset already called");
-        _resetCalled = true;
+    function setAccrualWindows(uint256[] calldata windows) external override onlyOwner {
+        require(windows.length % 2 == 0, "Invalid window length");
```

Comment:
i think we should limit the length of accrual windows, maybe something like:
* only whole numbers 0 to 23
* windows are 1 hour long, i + 1
* max 2 windows

possible bug vector: `windows` gets set with some stupidly high number of parameters like 0,1,0,1,0,1, ...
then when we do the check `isAccrualWindow` there's a stupidly high gas cost
(this can be reset by owner though so prob not a big deal)

Reply:
I think the first bullet point is already in place (24 allowed for the final number because the window is non-inclusive at the end)

Can add the second and third restrictions. But for fwiw `[0,1,0,1]` wouldn't be allowed bc the window values also have to be monotonically increasing